tools/check_spelling.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+
+# Applies requisite code formatters to the source tree
+# check_spelling.sh
+
+set -e
+
+VERSION="0.3.4"
+OS=""
+
+MISSPELL_ARGS="-error -o stderr"
+
+if [[ $1 == "fix" ]];then
+  MISSPELL_ARGS="-w"
+fi
+
+if [[ `uname` == "Darwin" ]];then
+  OS="mac"
+elif [[ `uname` == "Linux" ]];then
+  OS="linux"
+else
+  echo "Current only support mac/Linux"
+  exit 1
+fi
+
+SCRIPTPATH=$( cd "$(dirname "$0")" ; pwd -P )
+ROOTDIR=$SCRIPTPATH/..
+cd "$ROOTDIR"
+
+# Install tools we need
+if [[ ! -e "/tmp/misspell"  ]];then
+  wget -qO- https://github.com/client9/misspell/releases/download/v${VERSION}/misspell_${VERSION}_${OS}_64bit.tar.gz|tar xvz -C /tmp &> /dev/null
```

Comment:
You can remove the "v" from the "xvz" arg to tar. Then also you can remove the redirect to /dev/null, which might eat a transient error from wget that we might need for debugging.

Reply:
Also I would split the command here rather than piping it.  And do you really need `-q0` here? I'm actually not even sure what that does but in CI I see a ton of transient network issues cause failures, and I'd rather have default logging.

Reply:
`-O-` is to print the downloaded content to standard output, I am doing this to avoid creating intermediate files

Reply:
I'd never seen that mashed into the '-q'.  So can you get rid of the '-q'?

And why do you want to avoid an intermediate file?  You can put it in /tmp.

Reply:
To be honest, I don't have enough reason to avoid using intermediate files, I just hope can save a few steps

Reply:
My primary concern is that I expect the wget to fail occasionally or die halfway, and people will have to debug, so I don't want to lose its output. So my suggestion is to pull down the file as atomically as possible, without squelching diagnostic output.

Once the file is down, tar will be silent.

Reply:
I agree with you, I re-adjusted the code, output nothing in the normal case of wget, only print the error message when the error occurs, and increase the retry mechanism.